app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,50 @@
+class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+    if params[:question]
+      @question = Question.find_by(id: params[:question])
```

Comment:
You could set a single variable @commentable rather than question or answer. then in the view do
```
     <%= f.hidden_field :commentable_id, value: @commentable.id %>
     <%= f.hidden_field :commentable_type, value: @commentable.class.name %>
```